### DISCUSSION TITLE
Plot/Image Single-Animal Labeled Skeletons

### DISCUSSION THREAD

**Discussion by djg9324** [talmolab/sleap#1584] [Help!] [answered]
I am hoping to grab a .avi video of a single animal that has already been tracked and predicted on SLEAP and plot an imaged skeleton of the labels we have for it. We also already have a model that works with this video, so I hope that helps. The goal is to show the image of this skeleton but also at three different time points/frames (t=0, t=1, t=2). Additionally, I would like to plot this where there is no background or even the animal... just the labeled skeleton itself. I attached two images that help to illustrate what I am hoping to create. Can anybody help with this?
![sleap_skeleton_example](https://github.com/talmolab/sleap/assets/139586492/063dd236-989b-4e93-9f05-ac3462545b49)
![MoSeq example](https://github.com/talmolab/sleap/assets/139586492/5ae5991e-60a2-463f-9839-0f176eb9da3a)

**Comment by talmo** [comment, marked as answer]
Hi @djg9324,

Apologies for the delay!

Sure thing, there's a few different ways you can go about this, but the very simplest is to just plot the image like:

```py
import sleap

labels = sleap.load_file("my_labels.slp")

labels[0].plot()
```

This will plot the skeleton of all instances in the frame overlaid on the image.

If you want to plot it without the image, you'll need to get a bit more into the weeds, but one helper function in SLEAP might help:

```py
import sleap
import matplotlib.pyplot as plt

labels = sleap.load_file("my_labels.slp")

# Pull out a labeled frame.
lf = labels[0]

# Create a figure.
plt.figure(figsize=(6, 6))

# Plot instances in the frame.
for instance in lf.instances:
    sleap.nn.viz.plot_instance(instance)

# Set the axis bounds so you have the full frame in view and don't break aspect ratio.
plt.xlim([0, 1024])
plt.ylim([1024, 0])
```

You can check out the implementation of the [`plot_instance`](https://github.com/talmolab/sleap/blob/cb82d364e95f910e1306483369b8f0ef89998b86/sleap/nn/viz.py#L192-L255) function to see what's going on behind the scenes. You can probably customize it for your own needs pretty easily.

Let us know if that helps or if you have any questions!

Cheers,

Talmo
